kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/FilterIT.java

```diff
@@ -102,6 +108,118 @@ class FilterIT {
+        var config = proxy(cluster)
+                .addToFilterDefinitions(namedFilterDefinition)
+                .addToDefaultFilters(namedFilterDefinition.name());
+
+        try (var tester = kroxyliciousTester(config);
+                var client = tester.simpleTestClient()) {
+            MetadataRequestData message = new MetadataRequestData();
+            Uuid nonexistentTopic = Uuid.randomUuid();
+            message.unknownTaggedFields().add(
+                    new RawTaggedField(TopicIdToNameResponseStamper.TOPIC_ID_TAG, nonexistentTopic.toString().getBytes(StandardCharsets.UTF_8)));
+            Response response = client.getSync(new Request(METADATA, METADATA.latestVersion(), "client", message));
+            List<String> tags = unknownTaggedFieldsToStrings(response.payload().message(), TopicIdToNameResponseStamper.TOPIC_NAME_TAG).toList();
+            assertThat(tags).hasSize(1);
+            String tag = tags.getFirst();
+            assertThat(tag).isEqualTo(topicNameMapping(nonexistentTopic, null, "UNKNOWN_TOPIC_ID"));
+        }
+    }
+
+    @Test
+    void filtersCanLookUpPartiallyExistingTopics(KafkaCluster cluster, Topic topic1, Admin admin) throws Exception {
+        Map<String, TopicDescription> descriptionMap = admin.describeTopics(List.of(topic1.name())).allTopicNames().get(5, TimeUnit.SECONDS);
+        Uuid topic1Id = descriptionMap.get(topic1.name()).topicId();
+
+        NamedFilterDefinition namedFilterDefinition = new NamedFilterDefinitionBuilder("topicIdLookup",
+                TopicIdToNameResponseStamper.class.getName())
+                .build();
+        var config = proxy(cluster)
+                .addToFilterDefinitions(namedFilterDefinition)
+                .addToDefaultFilters(namedFilterDefinition.name());
+
+        try (var tester = kroxyliciousTester(config);
+                var client = tester.simpleTestClient()) {
+            MetadataRequestData message = new MetadataRequestData();
+            Uuid nonexistentTopic = Uuid.randomUuid();
+            message.unknownTaggedFields().add(
+                    new RawTaggedField(TopicIdToNameResponseStamper.TOPIC_ID_TAG, (nonexistentTopic + "," + topic1Id.toString()).getBytes(StandardCharsets.UTF_8)));
+            Response response = client.getSync(new Request(METADATA, METADATA.latestVersion(), "client", message));
+            List<String> tags = unknownTaggedFieldsToStrings(response.payload().message(), TopicIdToNameResponseStamper.TOPIC_NAME_TAG).toList();
+            assertThat(tags).hasSize(1);
+            String tag = tags.getFirst();
+            String[] topicNames = tag.split(",");
+            assertThat(topicNames).containsExactlyInAnyOrder(topicNameMapping(topic1Id, topic1.name(), null),
+                    topicNameMapping(nonexistentTopic, null, "UNKNOWN_TOPIC_ID"));
+        }
+    }
+
+    @Test
+    void topicNameLookupComposesWithOtherFilters(KafkaCluster cluster, Topic topic1, Topic topic2, Admin admin) throws Exception {
```

Comment:
Don't you want two versions of this test? One where the chain is topicIdLookup, topicNamePrefixer, showing the topicIdLookup sees prefixed names, and a second one where the chain is topicNamePrefixer, topicIdLookup should that topicIdLookup see unprefixed names.